src/test/java/teammates/logic/core/FeedbackSessionsLogicTest.java

```diff
@@ -700,6 +704,162 @@ public void testUpdateFeedbackSession_shouldAdjustEmailSendingStatusAccordingly(
     }
 
+    @Test
+    public void testUpdateFeedbackSessionsInstructorDeadlinesWithNewEmail() {
+        InstructorAttributes instructorToBeUpdated = dataBundle.instructors.get("instructor1OfCourse1");
+        String courseId = instructorToBeUpdated.getCourseId();
+        String oldEmailAddress = instructorToBeUpdated.getEmail();
+        String newEmailAddress = "new@email.tmt";
+
+        ______TS("Update email; transfers deadlines to new email.");
+
+        Map<Instant, Long> oldDeadlineCounts = fsLogic.getFeedbackSessionsForCourse(courseId)
+                .stream()
+                .map(FeedbackSessionAttributes::getInstructorDeadlines)
+                .filter(instructorDeadlines -> instructorDeadlines.containsKey(oldEmailAddress))
+                .map(instructorDeadlines -> instructorDeadlines.get(oldEmailAddress))
+                .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
+        assertEquals(2L, oldDeadlineCounts.values()
+                .stream()
+                .reduce(0L, Long::sum)
+                .longValue());
+
+        fsLogic.updateFeedbackSessionsInstructorDeadlinesWithNewEmail(courseId, oldEmailAddress, newEmailAddress);
+
+        assertTrue(fsLogic.getFeedbackSessionsForCourse(courseId)
+                .stream()
+                .noneMatch(feedbackSessionAttributes -> feedbackSessionAttributes.getInstructorDeadlines()
+                        .containsKey(oldEmailAddress)));
+        Map<Instant, Long> newDeadlineCounts = fsLogic.getFeedbackSessionsForCourse(courseId)
+                .stream()
+                .map(FeedbackSessionAttributes::getInstructorDeadlines)
+                .filter(instructorDeadlines -> instructorDeadlines.containsKey(newEmailAddress))
+                .map(instructorDeadlines -> instructorDeadlines.get(newEmailAddress))
+                .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
+        assertEquals(oldDeadlineCounts, newDeadlineCounts);
+    }
+
+    @Test
+    public void testUpdateFeedbackSessionsStudentDeadlinesWithNewEmail() {
+        StudentAttributes student4InCourse1 = dataBundle.students.get("student4InCourse1");
+        String courseId = student4InCourse1.getCourse();
+        String oldEmailAddress = student4InCourse1.getEmail();
+        String newEmailAddress = "new@email.tmt";
+
+        ______TS("Update email; transfers deadlines to new email.");
+
+        Map<Instant, Long> oldDeadlineCounts = fsLogic.getFeedbackSessionsForCourse(courseId)
+                .stream()
+                .map(FeedbackSessionAttributes::getStudentDeadlines)
+                .filter(studentDeadlines -> studentDeadlines.containsKey(oldEmailAddress))
+                .map(studentDeadlines -> studentDeadlines.get(oldEmailAddress))
+                .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
+        assertEquals(2L, oldDeadlineCounts.values()
```

Comment:
Indentation seems off where the sub stream is mixed up with the parent stream. Also, there's way to make the count remain in int by not using .counting()

Reply:
I used `summingInt` initially but I thought `counting` was clearer, just that the type had to be changed, but I thought this was not a problem. I can switch back to `summingInt` though.

Reply:
Just some thoughts... It seems ok looking on a second look. Just keep it this way

Reply:
Oops, I changed it already. I hope that's fine.